contracts/vaults/Claim.sol

```diff
@@ -93,29 +108,34 @@ contract Claim is Base {
         lpToken.safeTransfer(submittedClaim.beneficiary, claimBounty.hacker);
         lpToken.safeTransfer(submittedClaim.committee, claimBounty.committee);
         //storing the amount of token which can be swap and burned so it could be swapAndBurn in a separate tx.
-        swapAndBurns[_pid] += claimBounty.swapAndBurn;
-        governanceHatRewards[_pid] += claimBounty.governanceHat;
-        hackersHatRewards[submittedClaim.beneficiary][_pid] += claimBounty.hackerHatVested;
-
-        emit ApproveClaim(_pid,
-                        msg.sender,
-                        submittedClaim.beneficiary,
-                        submittedClaim.bountyPercentage,
-                        tokenLock,
-                        claimBounty);
-        assert(poolInfos[_pid].balance > 0);
+        swapAndBurns[pid] += claimBounty.swapAndBurn;
+        governanceHatRewards[pid] += claimBounty.governanceHat;
+        hackersHatRewards[submittedClaim.beneficiary][pid] += claimBounty.hackerHatVested;
+        delete activeClaims[pid];
+        delete submittedClaims[_claimId];
+        emit ApproveClaim(pid,
+            _claimId,
+            msg.sender,
+            submittedClaim.beneficiary,
+            submittedClaim.bountyPercentage,
+            tokenLock,
+            claimBounty);
+        assert(poolInfos[pid].balance > 0);
     }
 
     /**
-      * @notice Dismiss a claim for a bounty submitted by a committee.
+    * @notice Dismiss a claim for a bounty submitted by a committee.
     * Called either by Hats governance, or by anyone if the claim is over 5 weeks old.
-    * @param _pid The pool id
+    * @param _claimId The claim ID
     */
-    function dismissClaim(uint256 _pid) external {
+    function dismissClaim(uint256 _claimId) external {
+        uint256 pid = submittedClaims[_claimId].pid;
         // solhint-disable-next-line not-rely-on-time
-        require(msg.sender == owner() || submittedClaims[_pid].createdAt + 5 weeks < block.timestamp, "HVE09");
-        delete submittedClaims[_pid];
-        emit DismissClaim(_pid);
+        require(msg.sender == owner() || submittedClaims[_claimId].createdAt + 5 weeks < block.timestamp, "HVE09");
+        require(submittedClaims[_claimId].beneficiary != address(0), "HVE10");
```

Comment:
So this check is not really necessary, bc if beneficiary = 0, then the claim was already dismissed or never existed in the firsts place.
That said, I like to have an explicit check here, so let's just keep it

Reply:
i think we have to check because if we dont then if this is submitted with an invalid claimId it would dismiss the claim in pool 0..